duckdb-vortex/src/vortex_scan.cpp

```diff
@@ -223,26 +230,37 @@ static unique_ptr<VortexFileReader> OpenFileAndVerify(FileSystem &fs, const std:
 }
 
 static void CreateScanPartitions(ClientContext &context, const VortexBindData &bind,
-                                 VortexScanGlobalState &global_state) {
-	uint64_t file_idx = 0;
-	for (const auto &file_name : global_state.expanded_files) {
-		auto file_reader = OpenFileAndVerify(FileSystem::GetFileSystem(context), file_name, bind);
-
-		const uint64_t row_count = Try([&](auto err) { return vx_file_row_count(file_reader->file, err); });
-		const auto partition_count = std::max(static_cast<uint64_t>(1), row_count / PARTITION_SIZE);
-
-		for (uint64_t partition_idx = 0; partition_idx < partition_count; ++partition_idx) {
-			global_state.scan_partitions.push_back(VortexScanPartition {
-			    .file_idx = file_idx,
-			    .start_row = partition_idx * PARTITION_SIZE,
-			    .end_row = (partition_idx + 1) * PARTITION_SIZE,
-			});
-		}
-
-		global_state.scan_partitions.back().end_row = row_count;
-
-		++file_idx;
+                                 VortexScanGlobalState &global_state, VortexScanLocalState &local_state,
+                                 uint64_t file_idx, unique_ptr<VortexFileReader> &file_reader) {
+	const auto file_name = global_state.expanded_files[file_idx];
+	const auto row_count = Try([&](auto err) { return vx_file_row_count(file_reader->file, err); });
+
+	const auto thread_count = std::thread::hardware_concurrency();
+	const auto file_count = global_state.expanded_files.size();
+
+	// This is a multiple of the 2048 DuckDB vector size:
+	//
+	// Factors to consider:
+	//  1. A smaller value means more work for the Vortex file reader.
+	//  2. A larger value reduces the parallelism available to the scanner
+	const uint64_t partition_size = 2048 * (thread_count > file_count ? 32 : 64);
```

Comment:
Did this do anything?

Reply:
For clickbench this did give me a speedup. But my plan is to play around with this in isolation in separate PR/branch. The issue is that the behavior is very diff between running locally on a Mac and Intel machines. E.g. larger values have a large perf benefit on Mac, not so much on Intel.